lib/angular/sync-subscribers.js

```diff
@@ -0,0 +1,88 @@
+var config = require('../config');
+
+
+/**
+ * Subscribes to submission complete topic
+ * @param {Object} mediator used for communication
+ * @param {Object} resultSync - result sync manager
+ */
+function subscribeToSubmissionComplete(mediator, resultSync) {
+  mediator.subscribe('wfm:appform:submission:complete', function(event) {
+    var metaData = event.metaData.wfm;
+    var submissionResult = event.submissionResult;
+    resultSync.managerPromise
+      .then(function(manager) {
+        return manager.getByWorkorderId(metaData.workorderId)
+          .then(function(result) {
+            var stepResult = result.stepResults[metaData.step.code];
+            stepResult.submission = submissionResult;
+            return manager.update(result);
+          });
+      })
+      .then(function(result) {
+        mediator.publish('wfm:result:remote-update:' + result.workorderId, result);
+      });
+  });
+}
+
+
+/**
+ * Subscribes to sync update failed topic, updates result._syncStatus.
+ * @param {Object} mediator used for communication
+ * @param {Object} resultSync - result sync manager
+ */
+function subscribeToRemoteUpdateFailed(mediator, resultSync) {
+  mediator.subscribe('wfm:sync:remote_update_failed:' + config.datasetId, getSyncNotificationHandler(resultSync));
+}
+
+
+/**
+ * Subscribes to sync update applies topic, updates result._syncStatus.
+ * @param {Object} mediator used for communication
+ * @param {Object} resultSync - result sync manager
+ */
+function subscribeToRemoteUpdateApplied(mediator, resultSync) {
+  mediator.subscribe('wfm:sync:remote_update_applied:' + config.datasetId, getSyncNotificationHandler(resultSync));
+}
+
+
+/**
+ * Returns sync notification handler
+ * @param {object} resultSync - result sync manager
+ * @returns {Function} sync notification handler
+ */
+function getSyncNotificationHandler(resultSync) {
+  /**
+   * Handles sync topics and notifications, updates the result.syncStatus.
+   * @param {Object} syncNotification - notification from sync.
+   */
+  return function(syncNotification) {
+    resultSync.read(syncNotification.uid)
+      .then(function(result) {
+        result._syncStatus = result._syncStatus || {};
+        result._syncStatus.code = syncNotification.code;
+        result._syncStatus.message = syncNotification.message;
+        resultSync.manager.update(result);
+      });
+  };
+}
+
+
+/**
+ * Subscribes to all topics.
```

Comment:
Probably could be "Subscribing to topics published by other modules relevant to updating results."

"All topics" is a bit generic.